leaflet-geoman.d.ts

```diff
@@ -1,340 +1,779 @@
+
+
+        /****************************************** 
+        * 
+        * TODO: TRANSLATION EVENTS ON MAP ONLY
+        *
+        ********************************************/
+
+        /** Standard Leaflet event. Fired when any layer is removed. */
+        on(type: 'pm:langchange', fn: (e: { activeLang: string, oldLang: string, fallback: string, translations: PM.Translations }) => void): this;
+
+        /****************************************** 
+        * 
+        * TODO: CONTROL EVENTS ON MAP ONLY
+        *
+        ********************************************/
+
+        /** Fired when a Toolbar button is clicked. */
+        on(type: 'pm:buttonclick', fn: (e: { btnName: string, button: PM.Button }) => void): this;
+
+        /** Fired when a Toolbar action is clicked. */
+        on(type: 'pm:actionclick', fn: (e: { text: string; action: string; btnName: string; button: PM.Button }) => void): this;
+    }
+
+    namespace PM {
+        /** supported shape names. 'ImageOverlay' is in Edit Mode only. */
+        type SUPPORTED_SHAPES =
             | 'Marker'
             | 'Circle'
             | 'Line'
             | 'Rectangle'
             | 'Polygon'
             | 'Cut'
-            | 'CircleMarker';
-
-        export type GEOMAN_MAP_DRAW_MODE_EVENTS =
-            | 'pm:drawstart'
-            | 'pm:drawend'
-            | 'pm:create';
-
-        export type GEOMAN_MAP_EDIT_MODE_EVENTS = 'pm:globaleditmodetoggled';
-
-        export type GEOMAN_MAP_DRAG_MODE_EVENTS = 'pm:globaldragmodetoggled';
-
-        export type GEOMAN_MAP_REMOVAL_MODE_EVENTS = 'pm:globalremovalmodetoggled' | 'pm:remove' | 'layerremove';
-
-        export type GEOMAN_MAP_CUT_MODE_EVENTS = 'pm:cut' | 'pm:globalcutmodetoggled';
-
-        export type GEOMAN_LAYER_DRAW_MODE_EVENTS =
-            | 'pm:vertexadded'
-            | 'pm:snapdrag'
-            | 'pm:snap'
-            | 'pm:unsnap'
-            | 'pm:centerplaced';
-
-        export type GEOMAN_LAYER_EDIT_MODE_EVENTS =
-            | 'pm:edit'
-            | 'pm:update'
-            | 'pm:enable'
-            | 'pm:disable'
-            | 'pm:vertexadded'
-            | 'pm:vertexremoved'
-            | 'pm:vertexclick'
-            | 'pm:markerdragstart'
-            | 'pm:markerdrag'
-            | 'pm:markerdragend'
-            | 'pm:layerrest'
-            | 'pm:snapdrag'
-            | 'pm:snap'
-            | 'pm:unsnap'
-            | 'pm:intersect'
-            | 'pm:centerplaced';
-
-        export type GEOMAN_LAYER_DRAG_MODE_EVENTS =
-            | 'pm:dragstart'
-            | 'pm:drag'
-            | 'pm:dragend';
-
-        export type GEOMAN_LAYER_CUT_MODE_EVENTS = 'pm:cut' | 'pm:edit';
-
-        export interface GeomanHelpers {
-            getShapes(): string[];
-            getActiveShape(): string;
+            | 'CircleMarker'
+            | 'ImageOverlay';
+
+        /** supported shape types. 'ImageOverlay' is in Edit Mode only. */
+        type SUPPORTED_SHAPE_TYPES =
```

Comment:
"Are here the Leaflet types correct? L.Marker because we define them only as String this._shape = 'Line';"

This is mainly used for the `shape` property on the event parameters. Are you saying those aren't set to the actual leaflet shape and just contain the shape name as a string?